test/development/app-dir/owner-stack-invalid-element-type/owner-stack-invalid-element-type.test.ts

```diff
@@ -0,0 +1,147 @@
+import { nextTestSetup } from 'e2e-utils'
+import { assertHasRedbox, getRedboxSource } from 'next-test-utils'
+
+async function getStackFramesContent(browser) {
+  const stackFrameElements = await browser.elementsByCss(
+    '[data-nextjs-call-stack-frame]'
+  )
+  const stackFramesContent = (
+    await Promise.all(
+      stackFrameElements.map(async (frame) => {
+        const functionNameEl = await frame.$('[data-nextjs-frame-expanded]')
+        const sourceEl = await frame.$('[data-has-source]')
+        const functionName = functionNameEl
+          ? await functionNameEl.innerText()
+          : ''
+        const source = sourceEl ? await sourceEl.innerText() : ''
+
+        if (!functionName) {
+          return ''
+        }
+        return `at ${functionName} (${source})`
+      })
+    )
+  )
+    .filter(Boolean)
+    .join('\n')
+
+  return stackFramesContent
+}
+
+describe('app-dir - owner-stack-invalid-element-type', () => {
+  const { next } = nextTestSetup({
+    files: __dirname,
+  })
+
+  it('should catch invalid element from on client-only component', async () => {
+    const browser = await next.browser('/browser')
+
+    await assertHasRedbox(browser)
+    const source = await getRedboxSource(browser)
+
+    const stackFramesContent = await getStackFramesContent(browser)
+    if (process.env.TURBOPACK) {
+      expect(stackFramesContent).toMatchInlineSnapshot(
+        `"at Page (app/browser/page.js (10:10))"`
+      )
+      expect(source).toMatchInlineSnapshot(`
+        "app/browser/browser-only.js (8:7) @ BrowserOnly
```

Comment:
Why is the Component name here different between Turbopack and Webpack? Turbopack has `@ BrowserOnly`, Webpack `@ Foo`

Reply:
It's returned from the source map, retrieved by the line number and column number. There might be some inconsistence between two that we need to tackle later with sourcemap.

Reply:
>  two that we need to tackle later with sourcemap.

Then please leave a TODO or FIXME to highlight these issues. In this form, it's not clear what the intended behavior is.